Refactored Nodes1DProvisioner

k7miller opened the pull request:
- Updated documentation.
- Updated tests.
- Changed raw pointers to unique_ptr.
- Deleted copy constructor and copy assignment operator.
- Added default move constructor and move assignment operator.

k7miller committed:
Refactored Nodes1DProvisioner
- Updated documentation.
- Updated tests.
- Changed raw pointers to unique_ptr.
- Deleted copy constructor and copy assignment operator.
- Added default move constructor and move assignment operator.

mdunphy commented on src/test/Nodes1DProvisionerTests.cpp:
```diff
@@ -281,7 +281,7 @@ namespace blitzdg {
 				delete nodes1DProvisioner;
 			}
 
-			It(Should_Build_4th_Order_GradVandermonde_Matrix) {
```
Why kill this test?

k7miller replied:
We made the member function BuildGradVandermonde private since it doesn't need to be part of the public interface as far as I can tell.

mdunphy replied:
Fair enough

k7miller committed:
Fixed zero initialization of epart and npart in MeshManager

dsteinmo commented on include/Nodes1DProvisioner.hpp:
```diff
@@ -96,52 +146,29 @@ namespace blitzdg {
 
       /**
-       * Computes the Vandermonde matrix which maps modal coefficients to nodal values.
+       * Default move constructor.
```
If this is default, do we need to explicitly write it out here?

k7miller replied:
I can change the comment, but the defaulted move constructor definition needs to be part of the class definition.

dsteinmo commented on src/Nodes1DProvisioner.cpp:
```diff
@@ -274,7 +274,7 @@ namespace blitzdg {
 
     void Nodes1DProvisioner::buildVandermondeMatrix() {
-        V = new real_matrix_type(NOrder+1, NOrder+1);
+        V.reset(new real_matrix_type(NOrder + 1, NOrder + 1));
```
why reset this, can't we initialize in the initializer list?

k7miller replied:
Yes, it would be better in the initializer list. This is how the code was originally formatted, but I should have changed it.

dsteinmo commented on src/Nodes1DProvisioner.cpp:
```diff
@@ -284,14 +284,14 @@ namespace blitzdg {
         secondIndex jj;
 
-        Dr = new real_matrix_type(NOrder+1, NOrder+1);
+        Dr.reset(new real_matrix_type(NOrder + 1, NOrder + 1));
```
same comment applies, move to initializer?

k7miller replied:
Same as above.

dsteinmo commented on src/test/Nodes1DProvisionerTests.cpp:
```diff
@@ -281,7 +281,7 @@ namespace blitzdg {
 			}
 
-			It(Should_Build_4th_Order_GradVandermonde_Matrix) {
+			/* It(Should_Build_4th_Order_GradVandermonde_Matrix) {
```
Don't leave dead code in comments, just delete it.

k7miller replied:
Yah, good call.

k7miller committed:
Addressed pull request comments
- Updated Nodes1DProvisioner documentation.
- Initialized V and Dr in initializer list.
- Deleted commented out test in Nodes1DProvisionerTests.
- Updated norms in LinAlgHelpers to be templated on the variable type T.
- Added normFro to LinAlgHelpers to compute the Frobenius norm of a matrix.
- Fixed bugs in Nodes1DProvisionerTests related to computing the norm of the residual.

dsteinmo commented:
:shipit:
